gspread/models.py

```diff
@@ -288,6 +288,31 @@ def get_worksheet(self, index):
             return Worksheet(self, properties)
         except (KeyError, IndexError):
             return None
+    
+    def get_worksheet_by_id(self, id):
+        """Returns a worksheet with specified `worksheet id`.
+
+        :param id: The id of a worksheet. it can be seen in the url as the value of the parameter 'gid'.
+        :type id: int
+
+        :returns: an instance of :class:`gspread.models.Worksheet`
+                  or `None` if the worksheet is not found.
+
+        Example. To get first worksheet of a spreadsheet:
+
+        >>> sht = client.open('My fancy spreadsheet')
+        >>> worksheet = sht.get_worksheet_by_id(0):rtype: dict
```

Comment:
Minor detail, could you use an example where we access a sheet that is not the first one please ? the first sheet can be obtained using `sht.sheet1` as a shortcut, so the most common use of this function will be to access another sheet.

Reply:
He used `id=0` just to show how it would work, but the gid on the URL is often a big number, so it would become something like `sht.get_worksheet_by_id(2146239925)` for example. Instead of the user/developer use the method to get the worksheet by it's name, which he would need to copy its name or already know it, he's just need to copy the gid from url.

Reply:
Hi @JoachimKoenigslieb , thank you for reviewing this PR.

I understand it is an example. :slightly_smiling_face: 

My point is:
the provided example does not reflect how the user is most likely going to use the method. We already have a shortcut to access the first sheet (using: `sht.sheet1`). So the user is most likely going to access another sheet, like the second one or the _Nth_ one. This is why I recomend to update the example.

PS: I did not noticed that sheet ID are not incremental (1st => `0`, 2nd => `1`, etc), but it is a big number with no relation one from another. Having the example reflecting this (using a example sheet ID like `123456789`) would have teach me this detail :wink: